code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt

```diff
@@ -0,0 +1,240 @@
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateConstants.IntentActions.TIMER_EXPIRED
+import com.adobe.marketing.mobile.notificationbuilder.R
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setNotificationBackgroundColor
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setNotificationBodyTextColor
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setNotificationTitleTextColor
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.AEPPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.BasicPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockAEPPushTemplateDataProvider
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.replaceValueInMap
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.IntentData
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.MapData
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNotNull
+import junit.framework.TestCase.assertNull
+import org.junit.Assert.assertArrayEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.verify
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class AEPPushNotificationBuilderTest {
+
+    @Mock
+    private lateinit var smallLayout: RemoteViews
+
+    @Mock
+    private lateinit var expandedLayout: RemoteViews
+
+    private lateinit var context: Context
+    private lateinit var dataMap: MutableMap<String, String>
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var mockBundle: Bundle
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+        context = RuntimeEnvironment.getApplication()
+        dataMap = MockAEPPushTemplateDataProvider.getMockedAEPDataMapWithAllKeys()
+        trackerActivityClass = DummyActivity::class.java
+        mockBundle = MockAEPPushTemplateDataProvider.getMockedAEPBundleWithAllKeys()
+    }
+
+    @Test
+    fun `verify construct should map valid data fields to notification data`() {
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+        val notification = AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            CHANNEL_ID_TO_USE,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            CONTAINER_LAYOUT_VIEW_ID
+        ).build()
+
+        verifyNotificationDataFields(notification, pushTemplate)
+    }
+
+    @Test
+    fun `verify construct should map valid data fields and colors to notification view`() {
```

Comment:
nit: Any reason not to club this test and the one before this?

Reply:
no reason just wanted to keep notification-related tests and notification view-related tests separate, if you want I can combine them into one, what do you suggest?

Reply:
Don't really have a preference. Noticed that it was combined for the subsequent test that uses intent. No issue with this :)

Reply:
Yeah, that was intentional, I wanted to keep view related tests separate for better readability, however once it was validated for map then it made sense to combine it for the intent method

will proceed with this pattern